### PR TITLE
Sort the branches alphabetically when writing Frames

### PR DESCRIPTION
**Pull request by tmadlener**
BEGINRELEASENOTES
- Make the collections appear in alphabetical order in root files, using a case insensitive sorting of the collections that are written.

ENDRELEASENOTES

As "requested" in https://github.com/AIDASoft/DD4hep/pull/1108#issuecomment-1546252574

### CODE REVIEW

**Review by jmcarcell — APPROVED** [review]
Looks good, for consistency a similar thing should be done for the RNTuples

File: src/ROOTFrameWriter.cc [hunk 1]
```diff
@@ -18,14 +21,37 @@ void ROOTFrameWriter::writeFrame(const podio::Frame& frame, const std::string& c
...
 }
 
+/**
+ * Sort the input vector of strings alphabetically, case insensitive.
+ *
+ * NOTE: This creates a copy of the input vector, so don't use this in tight
+ * loops!
+ */
+std::vector<std::string> sortAlphabeticaly(const std::vector<std::string>& input) {
+  auto output = input;
```
[the remark this answers is not in the document]
Comment (tmadlener) [reply src/ROOTFrameWriter.cc:31]: Ah, yes. Excellent proposal. I always forget that this is a possibility for some reason.

**Comment by tmadlener** [comment]
I agree on the RNTuple. Would it help to move the `sortAlphabetically` to `root_utils` already now, so that it is easier to reuse?

**Comment by jmcarcell** [comment]
Yes, already done :)

File: src/ROOTFrameWriter.cc (hunk 1 above)
[the remark this answers is not in the document]
Comment (jmcarcell) [reply src/ROOTFrameWriter.cc]: For the name change it's only to make it clear that it's not modifying the input although that can be seen in the argument; I don't have a strong opinion

File: src/rootUtils.h [hunk 2]
```diff
@@ -152,6 +154,25 @@ inline auto reconstructCollectionInfo(TTree* eventTree, podio::CollectionIDTable
   return collInfo;
 }
 
+/**
+ * Sort the input vector of strings alphabetically, case insensitive.
+ */
+inline std::vector<std::string> sortAlphabeticaly(std::vector<std::string> strings) {
```
Comment (tmadlener) [inline src/rootUtils.h:160]:
```suggestion
inline std::vector<std::string> sortAlphabetically(std::vector<std::string> strings) {
```
proper english

File: src/ROOTFrameWriter.cc [hunk 3]
```diff
@@ -25,7 +25,7 @@ void ROOTFrameWriter::writeFrame(const podio::Frame& frame, const std::string& c
   // been initialized
   if (catInfo.tree == nullptr) {
     catInfo.idTable = frame.getCollectionIDTableForWrite();
-    catInfo.collsToWrite = collsToWrite;
+    catInfo.collsToWrite = root_utils::sortAlphabeticaly(collsToWrite);
```
Comment (tmadlener) [inline src/ROOTFrameWriter.cc:28]:
```suggestion
    catInfo.collsToWrite = root_utils::sortAlphabetically(collsToWrite);
```
same

**Comment by tmadlener** [comment]
Waiting for #423 to be merged first, just to make sure that this doesn't break things accidentally.

**Pull request merged**